kernel/src/hil/storage_interface.rs

```diff
@@ -0,0 +1,74 @@
+//! Interfaces for storage devices.
+
+use crate::returncode::ReturnCode;
+
+pub type StorageLen = usize;
+pub type OperationResult = Result<(), (ReturnCode, Option<&'static mut [u8]>)>;
```

Comment:
I would say remove these. We don't do this elsewhere in Tock HILs (I don't think).